bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisBoxHandler.java

```diff
@@ -0,0 +1,371 @@
+                    intesisBoxSocketApi.sendLimitsQuery();
+                    intesisBoxSocketApi.sendAlive();
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                }
+                pollingTask = scheduler.scheduleWithFixedDelay(this::polling, 3, 45, TimeUnit.SECONDS);
+                updateStatus(ThingStatus.ONLINE);
+            });
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No IP address specified)");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> pollingTask = this.pollingTask;
+
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+
+        if (pollingTask != null) {
+            pollingTask.cancel(true);
+            this.pollingTask = null;
+        }
+        if (api != null) {
+            api.closeConnection();
+        }
+        super.dispose();
+    }
+
+    private synchronized void polling() {
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+        if (api != null) {
+            if (!api.isConnected()) {
+                try {
+                    api.openConnection();
+                } catch (IOException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                }
+            }
+            api.sendAlive();
+            ;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        IntesisBoxSocketApi api = this.intesisBoxSocketApi;
+        if (api != null) {
+            if (!api.isConnected()) {
+                logger.trace("Sending command failed, not connected");
+                return;
+            }
+            if (command instanceof RefreshType) {
+                logger.warn("Refresh channel {}", channelUID.getId());
+                api.sendQuery(channelUID.getId());
+                return;
+            }
+        }
+        String value = "";
+        switch (channelUID.getId()) {
+            case CHANNEL_TYPE_POWER:
+                if (command instanceof OnOffType) {
+                    value = ((OnOffType) command == OnOffType.ON) ? "ON" : "OFF";
+                }
+                break;
+            case CHANNEL_TYPE_TARGETTEMP:
+                if (command instanceof QuantityType) {
+                    QuantityType<?> celsiusTemperature = (QuantityType<?>) command;
+                    celsiusTemperature = celsiusTemperature.toUnit(SIUnits.CELSIUS);
+                    if (celsiusTemperature != null) {
+                        double doubleValue = celsiusTemperature.doubleValue();
+                        if (doubleValue < minTemp) {
+                            doubleValue = minTemp;
+                        } else if (doubleValue > maxTemp) {
+                            doubleValue = maxTemp;
+                        }
```

Comment:
You could use the Math function.
```suggestion
                        doubleValue = Math.max(minTemp, Math.min(maxTemp, doubleValue));
```

Reply:
Done